<a href="https://colab.research.google.com/github/BramastaJaya/Basic-Python-Programs-in-Google-Colab/blob/main/Python's_Introduction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<div align="center">
<h1>Jurnal Praktikum Dasar Kecerdasan Artifisial</h1>
<h3>Modul: Pengenalan Python</h3>
</div>

### 1. Program Sederhana dengan Python (Bobot Nilai: `5`)

**a.** Buat program yang menampilkan kata "Hello, World!" pada terminal dengan menggunakan `print()`

In [ ]:
print("Hello, World!")

Hello, World!


*Test case* program

|**Input**|**Output**    |
|---------|--------------|
|         |'Hello, World!'|

**b.** Isi variabel rasio dengan akar dari 9 $(\sqrt{9})$.
Tip: untuk menghitung akar pangkat 2 dari $x$, dapat menggunakan x**(1/2)

In [ ]:
akar_sembilan = 9**0.5
print(akar_sembilan)

3.0


*Test case* program

|**Input**|**Output**    |
|---------|--------------|
|         |'3.0'|

**c.** Program di bawah ini menghitung nilai dari persamaan kuadrat $20x^2 + 25x + 30$ dengan meminta *input* dari pengguan untuk variabel $x$. Minta *input* pada pengguna dengan deskripsi `"Masukkan nilai x: "`, kemudian ubah variabel `x` menjadi `float`

In [ ]:
x = input("Masukkan nilai x: ")
x = float(x)

hasil_persamaan = 20*x**2 + 25*x + 30
print(hasil_persamaan)

Masukkan nilai x: 5
655.0


*Test case* program

|**Input**|**Output**    |
|---------|--------------|
|Masukkan nilai x: `3` |'285'|
|Masukkan nilai x: `4` |'450'|
|Masukkan nilai x: `5`|'655'|

### 2.  Pengkondisian dalam Python (Bobot Nilai: `10`)

**a.** Buat program untuk mengecek apakah suatu bilangan ganjil atau genap. Jika bilangan tersebut ganjil maka tampilkan `"Bilangan Ganjil"` pada terminal, sedangkan jika bilangan tersebut genap maka tampilkan `"Bilangan Genap"` pada terminal. Minta *input* pada pengguna dengan deskripsi (tidak ada ketentuan khusus untuk deskripsi), kemudian ubah variabel yang menampung hasil *input* pengguna menjadi `.integer`

In [ ]:
n = input("Masukkan sebuah angka: ")
n = int(n)

if n % 2 == 0:
    print("Bilangan Genap")
else:
    print("Bilangan Ganjil")

Masukkan sebuah angka: 5
Bilangan Ganjil


*Test case* program

|**Input**|**Output**    |
|---------|--------------|
|Masukkan nilai: `3` |'Bilangan Ganjil'|
|Masukkan nilai: `4` |'Bilangan Genap'|
|Masukkan nilai: `5`|'Bilangan Ganjil'|

**b.** Buat program untuk meminta *input* nilai dari pengguna dalam rentang 0 hingga 100. Program tersebut menerima nilai dan menampilkan indeks huruf dari nilai tersebut pada terminal sesuai dengan kriteria berikut
- **A**: 85 - 100
- **B**: 70 - 84
- **C**: 55 - 69
- **D**: 40 - 54
- **E**: 0 - 39

In [ ]:
nilai = input("Masukkan nilai 0 - 100: ")
nilai = int(nilai)
if nilai > 100:
  print("salah masukkan angka!")
elif nilai >= 85:
  print("Wow dapet A")
elif nilai >= 70:
  print("Eleh dapet B")
elif nilai >= 55:
  print("Dihh dapet C")
elif nilai >= 40:
  print("Astaga dapet D")
elif nilai >= 0:
  print("Serius dapet E?")
else:
  print("Anda salah memasukkan angka!")

Masukkan nilai 0 - 100: 42
Astaga dapet D


*Test case* program

|**Input**|**Output**    |
|---------|--------------|
|Masukkan nilai user: `91` |'A'|
|Masukkan nilai user: `75` |'B'|
|Masukkan nilai user: `42`|'D'|

### 3. Prosedur dan Fungsi (Bobot Nilai: `10`)

**a.** Buatlah sebuah prosedur yang menerima parameter berupa nama. Jika nama yang diberikan adalah `"Reza"`, prosedur akan mencetak output `"Selamat datang kembali, [nama]"`. Jika nama yang diberikan bukan "Reza", prosedur akan mencetak output `"Selamat datang, [nama]"`. Sesuaikan output berdasarkan nama yang diterima oleh parameter.

In [ ]:
def greet_reza(nama) :
    if nama == "Reza" :
        print("Selamat datang kembali, " + nama)
    else :
        print("Selamat datang, " + nama)

input_nama = input("Silahkan masukkan nama anda: ")
input_nama = str(input_nama)
greet_reza(input_nama)

Silahkan masukkan nama anda: Tiara
Selamat datang, Tiara


*Test case* program

|**Input**|**Output**    |
|---------|--------------|
|Masukkan nama: `Afeef` |'Selamat datang, Afeef'|
|Masukkan nama: `Reza` |'Selamat datang kembali, Reza'|
|Masukkan nama: `Tiara`|'Selamat datang, Tiara'|

**b.** Buatlah sebuah fungsi sederhana yang mengonversi jarak dari kilometer (KM) menjadi meter (M). Fungsi ini akan menerima input berupa jarak dalam kilometer, kemudian mengembalikan hasil konversi dalam meter.

In [ ]:
def converted(m):
    return m*1000

km = input("Masukkan jarak kilometer: ")
m = float(km)

hasil_konversi = int(converted(m))
print(km + "km sama dengan " + str(hasil_konversi) + "m")

Masukkan jarak kilometer: 5.5
5.5km sama dengan 5500m


*Test case* program

|**Input**|**Output**    |
|---------|--------------|
|Masukkan jarak (KM): `3` |'3000'|
|Masukkan jarak (KM): `4` |'4000'|
|Masukkan jarak (KM): `5.5`|'5500'|

### 4. Perulangan (Bobot Nilai: `15`)

**a.** Buatlah program yang menerima *input* dari pengguna berupa sebuah bilangan bulat positif `n`. Program tersebut harus menampiklan semua bilangan ganjil dari 1 hingga `n` menggunakan perulangan `for` dan fungsi `range()`. Top: gunakan langkah kelipatan 2 dalam `range()` untuk menampilkan hanya angka ganjil.

In [ ]:
value = input("Masukkan sebuah angka: ")
value = int(value)

for i in range(1, value+1, 2) :
    print(i)

Masukkan sebuah angka: 11
1
3
5
7
9
11


*Test case* program

|**Input**|**Output**    |
|---------|--------------|
|Masukkan n: `1` |'1'|
|Masukkan n: `6` |'1'<br>'1'<br>'5'|
|Masukkan n: `11`|'1'<br>'3'<br>'5'<br>'7'<br>'9'<br>'11'|

**b.** Buatlah sebuah fungsi yang menerima argumen `n` dan mengembalikan jumlah dari bilangan bulat dari 1 hingga n.  Sebagai contoh, jika `n = 5`, fungsi tersebut harus menghitung penjumlahan **$1+2+3+4+5$** dan mengembalikan nilai 15. Jika `n` kurang dari 1, kembalikan nilai 0. Gunakan perulangan `while` untuk menghitung jumlah tersebut.

In [ ]:
def increment(value):
    if value < 1 :
        return 0
    else :
        total = 0
        i = 1
        while i <= value:
            total += i
            i+=1
        return total

str_value = input("Masukkan angka: ")
int_value = int(str_value)

sum_result = increment(int_value)
print(str(int_value) + " = " + str(sum_result))

Masukkan angka: 10
10 = 55


*Test case* program

|**Input**|**Output**    |
|---------|--------------|
|Masukkan n: `5` |'15'|
|Masukkan n: `10` |'55'|
|Masukkan n: `-3`|'0'|

### 5. List dan Dictionary (Bobot Nilai: `20`)

**a.** Buat program yang mengganti ekstensi `.go` menjadi `.py` pada setiap elemen dalam list `filenames`. Gunakan metode `.replace()` untuk mengubah ekstensi, kemudian simpan nama file yang telah diubah ke dalam list baru bernama `new_filenames` dan kemudian cetak list ke dalam terminal.

Tip :
```python
sapa_tiara = "Hai Tiara!!"
print("Sapa Tiara:", sapa_tiara)

sapa_farah = sapa_tiara.replace("Tiara", "Farah")
print("Sapa Farah:", sapa_farah)
```

output
```
Sapa Tiara: Hai Tiara!!
Sapa Farah: Hai Farah!!
```

In [ ]:
filenames = ["app.go", "main.go", "formula.go", "program.go"]
new_filenames = []

for i in filenames :
    new_name = i.replace("go", "py")
    new_filenames.append(new_name)

print(new_filenames)

['app.py', 'main.py', 'formula.py', 'program.py']


*Test case* program

|**Input**|**Output**    |
|---------|--------------|
|         |['app.py', 'main.py', 'formula.py', 'program.py']|

Buatlah fungsi `email_list` yang menerima sebuah dictionary. Dictonary ini berisi nama domain sebagai `key`, dan daftar pengguna sebagai `value`. Lengkapi bagian yang kosong dalam fungsi untuk menghasilkan daftar yang berisi alamat email lengkap (misalnya, `dwiva.arya@gmail.com`). Alamat email ini dihasilkan dengan menggabungkan setiap nama pengguna dengan nama domain yang sesuai.

Tip:
```python
gender_dict = {
    "Rafly" : "Laki-Laki",
    "Iksan" : "Laki-Laki",
    "Imelda" : "Perempuan"
}

for key, value in gender_dict.items() :
    print("Key:", key, ", Value:", value)
```

Output
```
Key: Rafly , Value: Laki-Laki
Key: Iksan , Value: Laki-Laki
Key: Imelda , Value: Perempuan
```

In [ ]:
def email_list(domains):
    emails = []
    for i, j in domains.items():
        for user in j:
            emails.append(user+'@'+ i)

    return emails


email_domain = {
    "gmail.com"              : ["rafly.arj", "syahreza.adnan", "valen.hartanto"],
    "telkomuniversity.ac.id" : ["sabrina.tiara", "imelda"],
    "yahoo.com"              : ["raihan.omen", "dito.asyraf"]
}

dictionary_domain = email_list(email_domain)

print(dictionary_domain)


['rafly.arj@gmail.com', 'syahreza.adnan@gmail.com', 'valen.hartanto@gmail.com', 'sabrina.tiara@telkomuniversity.ac.id', 'imelda@telkomuniversity.ac.id', 'raihan.omen@yahoo.com', 'dito.asyraf@yahoo.com']


*Test case* program

|**Input**|**Output**    |
|---------|--------------|
|         |['rafly.arj@gmail.com', 'syahreza.adnan@gmail.com', 'valen.hartanto@gmail.com', 'sabrina.tiara@telkomuniversity.ac.id', 'imelda@telkomuniversity.ac.id', 'raihan.omen@yahoo.com', 'dito.asyraf@yahoo.com']|

### 6. Soal Cerita (Bobot Nilai: `40`)

Seorang guru ingin membuat program sederhana untuk mencatat nilai ujian siswa. Setiap siswa akan mengikuti **tiga mata pelajaran**. Guru perlu menghitung rata-rata nilai untuk setiap siswa dan menentukan apakah siswa tersebut **lulus** atau **tidak**. Kriteria kelulusan adalah jika rata-rata nilai siswa sama dengan atau lebih dair **70**.

Buatlah program yang melakukan hal berikut :
1. **Input/Output**
    - Meminta input dari pengguna berupa `nama siswa` dan `nilai` untuk tiga mata pelajaran
    - Menampilkan **rata-rata nilai** dan **status kelulusan** setiap siswa

2. **Penentuan kelulusan**
    - Menentukan apakah siswa **lulus** atau **tidak** berdasarkan rata-rata nilainya.

3. **Pengulangan program**
    - Menggunakan perulangan untuk memasukkan data beberapa siswa hingga pengguna memasukkan `"selesai"` sebagai nama siswa

4. **Perhitungan rata-rata**
    - Membuat fungsi untuk **menghitung rata-rata** yang menerima list nilai sebagai parameter dan mengembalikan rata-rata nilai dari list tersebut

5. **Penyimpanan data**
    - Gunakan *dictionary* dengan `keys` nama siswa, dan `value` list nilai dari siswa tersebut.

In [1]:
def add_key_value (data):
  while True:
    name = input("\nMasukkan nama siswa: ")
    if name.lower() == "selesai":
      break
    else:
      data[name] = []
      for j in range(0, 3):
        grade = input("Masukkan nilai siswa: ")
        grade = float(grade)
        data[name].append(grade)
  return data

student_data = {}
teacher = {}
database = add_key_value(student_data)

for i, j in database.items():
  teacher[i] = []
  total = 0
  total = float(total)
  add   = 0
  add   = float(add)
  for grade in j:
    total+=grade
    add+=1
  avg = total/add
  avg = float(avg)
  teacher[i] = [avg]

print("\nData Siswa:")
print(database)
for i,j in teacher.items():
  for status in j:
    status = int(status)
    if status >= 70:
      print("Rata-rata nilai " + str(i) + " adalah " + format(j) + ". Status: Lulus")
    else:
      print("Rata-rata nilai " + str(i) + " adalah " + format(j) + ". Status: Tidak Lulus")









Masukkan nama siswa: Andi
Masukkan nilai siswa: 80
Masukkan nilai siswa: 75
Masukkan nilai siswa: 90

Masukkan nama siswa: selesai

Data Siswa:
{'Andi': [80.0, 75.0, 90.0]}
Rata-rata nilai Andi adalah [81.66666666666667]. Status: Lulus


**Bagian yang diberikan tanda `ini` adalah masukkan user**

*Test case* program

**|----------------------------------------------------------------------------------|**

Masukkan nama siswa (atau ketik 'selesai' untuk berhenti): `Andi`

Masukkan nilai mata pelajaran 1: `80`

Masukkan nilai mata pelajaran 2: `75`

Masukkan nilai mata pelajaran 3: `90`

Rata-rata nilai Andi adalah 81.67. Status: Lulus

Masukkan nama siswa (atau ketik 'selesai' untuk berhenti): `Budi`

Masukkan nilai mata pelajaran 1: `60`

Masukkan nilai mata pelajaran 2: `65`

Masukkan nilai mata pelajaran 3: `70`

Rata-rata nilai Budi adalah 65.00. Status: Tidak Lulus

Masukkan nama siswa (atau ketik 'selesai' untuk berhenti): `selesai`

Data Siswa:
- Andi: [80, 75, 90], Rata-rata: 81.67, Status: Lulus
- Budi: [60, 65, 70], Rata-rata: 65.00, Status: Tidak Lulus

**|----------------------------------------------------------------------------------|**